ddl/db_test.go

```diff
@@ -2115,6 +2115,12 @@ LOOP:
 	hasNotNull = mysql.HasNotNullFlag(colC.Flag)
 	c.Assert(hasNotNull, IsFalse)
 
+	// add datetime type column
+	s.mustExec(tk, c, "create table test_on_update_e (c1 int);")
```

Comment:
No defer dropping?

Reply:
There are multiple `create` statements without defer dropping, e.g. https://github.com/pingcap/tidb/blob/322639bf25dfa54496d4f4bdd6a81fdc62c96b97/ddl/db_test.go#L2096. I  did not think much. Maybe it is because `txn.Rollback` at line https://github.com/pingcap/tidb/blob/322639bf25dfa54496d4f4bdd6a81fdc62c96b97/ddl/db_test.go#L2066 or a read-only storage backend?

Reply:
Rollback won't rollback DDL. Some other test cases contain dropping tables.
You'd better add dropping.

Reply:
Checked, no drop statements for other two tables. Added drop statements to these three tables.